### PR TITLE
Remove setup.py

### PR DESCRIPTION
**Pull request by xzmeng**

### CODE REVIEW

**Review by xmatthias — CHANGES_REQUESTED** [review]
I'm not per se against removing `setup.py` - but under the condition that we'll not lose functionality.

The current change will "lose" several subconditions ([dev], [all] - used as `pip install -e .[dev]` to test what i mean). While the commands won't fail - they'll not actually install the "extras" - just ignored as "this was a typo" i assume.

Also, it loses `tests_require` completely - which i think has special meaning in some scenarios - unless that's been deprecated/removed from setuptools (source?)

Please note: if supporting this the extras means "we need to duplicate all dependencies" - then it's not something i intend to accept (i couldn't figure this out, which is wthe main reason we still have the `setup.py` file).

**Comment by xzmeng** [comment]
The first problem can be solved with "recursive optional dependencies" since pip 21.2 supports it:
https://hynek.me/articles/python-recursive-optional-dependencies/

`tests_require` obtains the addtional packages when executing`python setup.py test`, and it's deprecated since setuptools version 41.5.0. We are using `pytest` interface in dev or ci, so it seems can be safely removed.

> tests_require
> If your project’s tests need one or more additional packages besides those needed to install it, you can use this option to specify them. It should be a string or list of strings specifying what other distributions need to be present for the package’s tests to run. When you run the test command, setuptools will attempt to obtain these. Note that these required projects will not be installed on the system where the tests are run, but only downloaded to the project’s setup directory if they’re not already installed locally.
> 
> Warning
> 
> Deprecated since version 41.5.0: The test command will be removed in a future version of setuptools, alongside any test configuration parameter.

https://setuptools.pypa.io/en/latest/references/keywords.html

File: pyproject.toml [hunk 1]
```diff
@@ -27,6 +27,111 @@ classifiers = [
...
+  "types-tabulate",
+]
+jupyter = [
+  "jupyter",
+  "nbstripout",
+  "ipykernel",
+  "nbconvert",
+]
+all = [
+  "freqtrade[plot,hyperopt,freqai,freqai_rl,hdf5,jupyter]",
```
Comment (xmatthias) [inline pyproject.toml:130]: previously, dev and all had the same dependencies.

I assume it's an intentional thought behind not having these aligned now? (i don't mind it at all - but want to know the why :laughing: )

Comment (xzmeng) [reply pyproject.toml:130]: Oh, I didn't notice the nuance, imo for the end users "all" is meant to install all feature dependencies, but not include development dependencies.

**Review by xmatthias — APPROVED** [review]
Nice enhancement :100: 

LGTM :+1:

**Pull request merged**